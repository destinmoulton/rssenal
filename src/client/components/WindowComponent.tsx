import * as React from "react";

import { Loader } from "semantic-ui-react";

import Browser from "./Browser";
import ErrorBoundary from "./ErrorBoundary";
import LoginContainer from "../containers/LoginContainer";

export interface IWindowComponentProps {
    isAuthorized?: boolean;
    isValidatingToken?: boolean;
    validateToken?: () => void;
}

class WindowComponent extends React.Component<IWindowComponentProps> {
    componentDidMount() {
        if (localStorage.getItem("jwt_token")) {
            this.props.validateToken();
        }
    }

    render() {
        const { isAuthorized, isValidatingToken } = this.props;

        let display = null;
        if (isAuthorized) {
            display = <Browser />;
        } else if (isValidatingToken) {
            display = <Loader />;
        } else {
            display = <LoginContainer />;
        }

        return <ErrorBoundary>{display}</ErrorBoundary>;
    }
}

export default WindowComponent;
