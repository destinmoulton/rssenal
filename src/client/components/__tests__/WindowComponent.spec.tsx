import * as React from "react";
import { shallow } from "enzyme";

import WindowComponent from "../WindowComponent";

describe("<WindowComponent />", () => {
    it("renders and matches snapshot", () => {
        const wrapper = shallow(<WindowComponent />);

        expect(wrapper).toMatchSnapshot();
    });
});
