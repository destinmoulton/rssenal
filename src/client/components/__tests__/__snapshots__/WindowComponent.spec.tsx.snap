// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WindowComponent /> renders and matches snapshot 1`] = `
<ErrorBoundary>
  <Connect(LoginComponent) />
</ErrorBoundary>
`;
